lib/membrane_rtsp/parser.ex

```diff
@@ -0,0 +1,112 @@
+defmodule Membrane.RTSP.Parser do
+  @moduledoc false
+
+  import NimbleParsec
+
+  @methods [
+    "DESCRIBE",
+    "ANNOUNCE",
+    "GET_PARAMETER",
+    "OPTIONS",
+    "PAUSE",
+    "PLAY",
+    "RECORD",
+    "REDIRECT",
+    "SETUP",
+    "SET_PARAMETER",
+    "TEARDOWN"
+  ]
+
+  alpha_num = ascii_string([?0..?9, ?a..?z, ?A..?Z, ?-, ?_], min: 1)
+  space = ascii_string([?\s, ?\t, ?\f], min: 1)
+  delimeter = choice(Enum.map(["\r\n", "\r", "\n"], &string/1))
+
+  ignored_space = ignore(space)
+  ignored_delimeter = ignore(delimeter)
+
+  methods = choice(Enum.map(@methods, &string/1))
+
+  header =
+    alpha_num
+    |> ignore(string(":"))
+    |> concat(ignored_space)
+    |> utf8_string([{:not, ?\r}, {:not, ?\n}], min: 1)
+    |> concat(ignored_delimeter)
```

Comment:
I'd get rid of the `ignored_` vars, as they actually seem to make the code longer.

Additionally, let's define `header_value = ...` together with `header_name`

```suggestion
  header =
    header_name
    |> ignore(string(":"))
    |> ignore(space)
    |> concat(header_value)
    |> ignore(delimiter)
```